app/src/androidTest/java/org/mozilla/fenix/ui/BadServerResponseTest.kt

```diff
@@ -0,0 +1,88 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.ui
+
+import okhttp3.mockwebserver.MockWebServer
+import org.junit.After
+import org.junit.Test
+import org.mozilla.fenix.helpers.HomeActivityTestRule
+import org.mozilla.fenix.helpers.MockWebServerHelper
+import org.mozilla.fenix.ui.robots.homeScreen
+
+class BadServerResponseTest {
```

Comment:
Is this test only run on debug builds? I remembered many of our SDKs are disabled in debug builds so this wouldn't catch their errors before they hit nightly. I think the solutions are 1) run this test on nightly or 2) enable the SDKs in debug, both of which seem like they might be outside the scope of this issue.

That being said, it'll be worth having this test for the connections we do make so I think we should keep going.

Reply:
On running the tests on Nightly, @isabelrios might have some ideas, as she is currently working on Nightly startup tests. I don't know if these would be worth/too risky to add (now or maybe in the future).

Reply:
We have tried to add tests on Nighly builds but we are having issues with the current decission task defined, there are problems with the way the Nightly build for the test is being signed and so the decission task is broken on master (see [PR](https://github.com/mozilla-mobile/fenix/pull/17948) for more info) Good news though is that there may be a solution that @escapewindow proposed in that PR that we are going to try. 

If that works, we can add this test to be run as part as the new task to run tests on Nighly (we will start by running just a single [test](https://github.com/mozilla-mobile/fenix/blob/master/automation/taskcluster/androidTest/flank-x86-start-test.yml#L37), and then add the ones like this that could be important to run specifically on particular builds)

Also, in case it helps, tests can be added to run on builds, in that case tests are running successfully on release branches (see [link](https://github.com/mozilla-mobile/fenix/runs/1960507602))